daemon/cmd/daemon_main.go

```diff
@@ -898,6 +898,9 @@ func init() {
 	flags.Int(option.LBMapEntriesName, lbmap.MaxEntries, "Maximum number of entries in Cilium BPF lbmap")
 	option.BindEnv(option.LBMapEntriesName)
 
+	flags.String(option.LocalRouterIP, "", "Link-local IP used for Cilium's router devices")
+	option.BindEnv(option.LocalRouterIP)
```

Comment:
Does this mean that you'll configure the same router IP on every node? Can you think of any potential repercussions with that? Like could we end up trying to use the cilium_host IP for inter-node traffic and the responses get routed back to the destination host instead of the host that originated the request?

Reply:
> Does this mean that you'll configure the same router IP on every node?

Yes.

> Can you think of any potential repercussions with that? Like could we end up trying to use the cilium_host IP for inter-node traffic and the responses get routed back to the destination host instead of the host that originated the request?

Hmm, I'll check if there's a situation where we use router IP to send packets, to provide a bit more context, here's some discussion on this over slack:
https://cilium.slack.com/archives/C2B917YHE/p1611824939057200?thread_ts=1611810661.056600&cid=C2B917YHE

Reply:
hmm,  I don't think `cilium_host` is used for routing purpose? But maybe I overlooked something? @joestringer is there anything specific you have in mind (like which mode would Cilium use `cilium_host` for routing purpose)?

Reply:
I was under the impression that sometimes when encryption is enabled, we use that IP as the source IP for cross-node traffic.

Reply:
The only reference to `cilium_host` seems to be https://github.com/cilium/cilium/blob/master/bpf/lib/encrypt.h#L74, which should only affect local routing, is this what you have in mind?

Reply:
> I was under the impression that sometimes when encryption is enabled, we use that IP as the source IP for cross-node traffic.

That's correct, at least in some cases. We use the `ciliumInternalIP`:
https://github.com/cilium/cilium/blob/7719c2dcaabfae23b9ec4ec27b35b3b03aa0c70b/pkg/datapath/linux/node.go#L829-L833
which can be (I'm not sure it always is) the `cilium_host` IP:
https://github.com/cilium/cilium/blob/7719c2dcaabfae23b9ec4ec27b35b3b03aa0c70b/pkg/node/types/node.go#L291-L293

Reply:
I see, seems like the following could also be using `cilium_host`'s IP for routing purpose: https://github.com/cilium/cilium/blob/master/pkg/datapath/iptables/iptables.go#L1004-L1023

Reply:
maybe we should scope this to very specific configs, like non-tunnel + non-ipsec + no-ipam?

Reply:
Added more config checks to only allow this option in non-tunnel + no-ipam + endpoint-routing + non-ipsec mode.